src/engine/Minus.cpp

```diff
@@ -95,6 +95,23 @@ size_t Minus::getCostEstimate() {
   return _left->getCostEstimate() + _right->getCostEstimate() + costEstimate;
 }
 
+// _____________________________________________________________________________
+auto Minus::makeUndefRangesChecker(bool left) const {
+  std::variant<ad_utility::Noop, ad_utility::FindSmallerUndefRanges> findUndef;
+  const auto& operation = left ? _left : _right;
+  for (const auto& cols : _matchedColumns) {
+    const auto& [_, info] =
+        operation->getVariableAndInfoByColumnIndex(cols[!left]);
+    // Use expensive operation if one of the columns might contain undef.
+    if (info.mightContainUndef_ !=
+        ColumnIndexAndTypeInfo::UndefStatus::AlwaysDefined) {
+      findUndef.emplace<1>();
+      break;
+    }
+  }
```

Comment:
This can be more idiomatic if you first compute the bool whether something can have undef
(e.g. use `ql::ranges::any_of` with a transformation.